src/node_file.cc

```diff
@@ -992,6 +993,23 @@ void Close(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+static void FastClose(Local<Object> recv,
+                      const int32_t fd,
+                      // NOLINTNEXTLINE(runtime/references) This is V8 api.
+                      v8::FastApiCallbackOptions& options) {
+  Environment* env = Environment::GetCurrent(recv->GetCreationContextChecked());
+
+  env->RemoveUnmanagedFd(fd);
+
+  uv_fs_t req;
+  FS_SYNC_TRACE_BEGIN(close);
+  options.fallback = uv_fs_close(nullptr, &req, fd, nullptr) < 0;
```

Comment:
Also, is not missing `GetValidatedFd` call here?

Reply:
GetValidatedFd checks for fd < 0 || fd > int32_max, which in this case will not even go into the fast path, because the type of fd is `in32_t`, which cannot be negative, and guaranteed by the v8 fast API that it won't enter into this function if it is int64_t for example.